instrumentation/async-http-client/async-http-client-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/v2_0/AsyncHttpClientNetAttributesGetter.java

```diff
@@ -21,13 +25,35 @@ public String getTransport(RequestContext request, @Nullable Response response)
 
   @Nullable
   @Override
-  public String getPeerName(RequestContext requestContext) {
-    return requestContext.getRequest().getUri().getHost();
+  public String getProtocolName(RequestContext request, @Nullable Response response) {
+    return Optional.of(request)
+        .map(RequestContext::getNettyRequest)
+        .map(NettyRequest::getHttpRequest)
+        .map(HttpMessage::getProtocolVersion)
+        .map(HttpVersion::protocolName)
+        .orElse(null);
```

Comment:
I tend to prefer not instantiating Optional/Stream here, but also we don't really have any established heuristic for when to use or not use Optional/Stream

Reply:
I used optional because it made the code simpler; but after thinking about it I agree that it's better to use `if`s here.

> but also we don't really have any established heuristic for when to use or not use Optional/Stream

Should we add sth like

> Don't use `Optional`/`Stream` in the hot path (instrumentation code), unless the instrumented library itself uses it.

to the contributing docs?